### PR TITLE
dbus: enable static build

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **dbus/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
ok, all builds pass on https://github.com/eirikb/proof-of-conan/actions/runs/5727727397

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/5741443485)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/5741547872)

File: recipes/dbus/1.x.x/conanfile.py [hunk 1]
```diff
@@ -21,10 +20,11 @@ class DbusConan(ConanFile):
     homepage = "https://www.freedesktop.org/wiki/Software/dbus"
     description = "D-Bus is a simple system for interprocess communication and coordination."
     topics = "bus", "interprocess", "message"
-    package_type = "shared-library"
```
[the remark this answers is not in the document]
Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:24]: I don't know. Is it fine if the package contains an executable too?

File: recipes/dbus/1.x.x/conanfile.py [hunk 2]
```diff
@@ -34,6 +34,8 @@ class DbusConan(ConanFile):
         "session_socket_dir": ["ANY"],
     }
     default_options = {
+        "shared": False,
```
[the remark this answers is not in the document]
Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:37]: I cannot think of an unsafe case. Do you have something in mind?

**Comment by ericLemanissier** [comment]
Let's go!

File: recipes/dbus/1.x.x/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jwillikers) [reply recipes/dbus/1.x.x/conanfile.py:37]: I mostly just wanted to double check. Thanks!

File: recipes/dbus/1.x.x/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwillikers) [reply recipes/dbus/1.x.x/conanfile.py:24]: Now that's a good a question. I guess I'm not sure. Maybe someone from the Conan team can tell us.

**Comment by ericLemanissier** [comment]
@uilianries @jcar87 @RubenRBS what do you think of this PR?

File: recipes/dbus/1.x.x/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/dbus/1.x.x/conanfile.py:24]: Yes, when a package is library + executable, the package_type = `library`

File: recipes/dbus/1.x.x/conanfile.py [hunk 3]
```diff
@@ -21,10 +20,11 @@ class DbusConan(ConanFile):
     homepage = "https://www.freedesktop.org/wiki/Software/dbus"
     description = "D-Bus is a simple system for interprocess communication and coordination."
     topics = "bus", "interprocess", "message"
```
Comment (uilianries) [inline recipes/dbus/1.x.x/conanfile.py:22]:
```suggestion
    topics = "bus", "interprocess", "message"
    package_type = "library"
```

**Comment by uilianries** [comment]
@ericLemanissier so nothing changed, only removed old versions?

**Comment by ericLemanissier** [comment]
The only other change is adding the shared option yes, which in the end will solve the missing binaries for conan 2 CI

**Comment by RubenRBS** [comment]
The permission issue should now be fixed, re-triggering :)

**Review by RubenRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by jwillikers** [comment]
Fyi, this removed the 1.14 dbus version, which was the last stable release. I just foun out that odd-numbered minor versions are development releases, see [here](https://gitlab.freedesktop.org/dbus/dbus/-/issues/494#note_2239032).

**Pull request merged**